src/core/src/sketch/minhash.rs

```diff
@@ -1356,13 +1366,14 @@ impl KmerMinHashBTree {
     }
 
     pub fn count_common(&self, other: &KmerMinHashBTree, downsample: bool) -> Result<u64, Error> {
-        if downsample && self.max_hash != other.max_hash {
-            let (first, second) = if self.max_hash < other.max_hash {
+        if downsample && self.scaled() != other.scaled() {
+            // downsample to the larger of the two scaled values
+            let (first, second) = if self.scaled() > other.scaled() {
                 (self, other)
             } else {
                 (other, self)
             };
-            let downsampled_mh = second.downsample_max_hash(first.max_hash)?;
+            let downsampled_mh = second.clone().downsample_scaled(first.scaled())?;
```

Comment:
I wonder if we can do (in a future PR, not this one) a new `.downsampled_iter(scaled)` for operations like `count_common`, and avoid the conversion.

The downsampled iter would iterate over values that are in the appropriate scaled value, but wouldn't need to create new minhash sketches (can reuse the largest one and stop returning values once they go over `max_hash`, for example)